windows/scripts/utils.ps1

```diff
@@ -516,3 +516,54 @@ function expand_template($template)
     }
     return $expanded_template
 }
+
+function dumpeventlog($path){
+	
+	Get-Eventlog -list | Where-Object { $_.Entries -ne '0' } | ForEach-Object {
+		$logFileName = $_.LogDisplayName
+		$exportFileName =$path + "\eventlog_" + $logFileName + ".evt"
+		$exportFileName = $exportFileName.replace(" ","_")
+		$logFile = Get-WmiObject Win32_NTEventlogFile | Where-Object {$_.logfilename -eq $logFileName}
+		try{
+			$logFile.backupeventlog($exportFileName)
+		} catch {
+			Write-Host "Could not dump $_.LogDisplayName (it might not exist)."
+		}
+	}
+}
+
+function exporteventlog($path){
+
+	Get-Eventlog -list | Where-Object { $_.Entries -ne '0' } | ForEach-Object {
```

Comment:
You may want to retrieve the event logs from specific services as well (get-eventlog won't retrieve those). For exampel, you could retrieve the logs for all the Hyper-V services (e.g. VMMS) or the WMI providers (e.g. get-winevent -filterhastable @{LogName="Hyper-V*"}